module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,656 @@
+                    i++;
+                }
+            }
+
+            {
+                std::cout << "  sm.gyros: " << std::endl;
+                int i = 0;
+                for (auto gyro : sensor_measurements.gyros) {
+                    std::cout << "    sm.gyros[" << i << "]" << std::endl;
+                    std::cout << "      name: " << gyro.name << std::endl;
+                    std::cout << "      value: [" << gyro.value.X << ", " << gyro.value.Y << ", " << gyro.value.Z << "]"
+                              << std::endl;
+                    i++;
+                }
+            }
+
+            {
+                std::cout << "  sm.position_sensors: " << std::endl;
+                int i = 0;
+                for (auto sensor : sensor_measurements.position_sensors) {
+                    std::cout << "    sm.position_sensors[" << i << "]" << std::endl;
+                    std::cout << "      name: " << sensor.name << std::endl;
+                    std::cout << "      value: " << sensor.value << std::endl;
+                    i++;
+                }
+            }
+        }
+
+        // Read each field of msg, translate it to our protobuf and emit the data
+        auto sensor_data = std::make_unique<DarwinSensors>();
+
+        sensor_data->timestamp = NUClear::clock::now();
+
+        for (const auto& position : sensor_measurements.position_sensors) {
+            translate_servo_id(position.name, sensor_data->servo).present_position = position.value;
+        }
+
+        if (sensor_measurements.accelerometers.size() > 0) {
+            // .accelerometers is a list of one, since our robots have only one accelerometer
+            const auto& accelerometer    = sensor_measurements.accelerometers[0];
+            sensor_data->accelerometer.x = static_cast<float>(accelerometer.value.X);
+            sensor_data->accelerometer.y = static_cast<float>(accelerometer.value.Y);
+            sensor_data->accelerometer.z = static_cast<float>(accelerometer.value.Z);
+        }
+
+        if (sensor_measurements.gyros.size() > 0) {
+            // .gyros is a list of one, since our robots have only one gyroscope
+            const auto& gyro         = sensor_measurements.gyros[0];
+            sensor_data->gyroscope.x = static_cast<float>(gyro.value.X);
+            sensor_data->gyroscope.y = static_cast<float>(gyro.value.Y);
+            sensor_data->gyroscope.z = static_cast<float>(gyro.value.Z);
+        }
+
+        // Ignore as we don't have physical functionality for this
+        /*
+        for (const auto& bumper : sensor_measurements.bumpers) {
+            // string name
+            // bool value
+        }
+
+        for (const auto& force_3d : sensor_measurements.force3ds) {
+            // string name
+            // Vector3 value
+        }
+
+        for (const auto& force_6d : sensor_measurements.force6ds) {
+            // string name
+            // Vector3 value
+        }
+        */
```

Comment:
Delete commented code? Pretty sure we will never have these? Unless FSRs manifest as bumpers?

Reply:
Pretty sure FSRs are bumpers.The rest can go